src/network/networkd-state-file.c

```diff
@@ -445,9 +468,13 @@ int link_save(Link *link) {
 
         if (link->network) {
                 char **dhcp6_domains = NULL, **dhcp_domains = NULL;
-                const char *dhcp_domainname = NULL, *p;
+                const char *dhcp_domainname = NULL, *online_state, *p;
                 bool space;
 
+                online_state = link_online_state_to_string(link->online_state);
+                if (online_state)
+                        fprintf(f, "ONLINE_STATE=%s\n", strempty(online_state));
```

Comment:
link_online_state_to_string() will return NULL if the state isn#t known/invalid, and a good non-empty string otherwise. hence you can drop the strempty() here

Reply:
Thanks, I'll address this in the next revision (pending some other changes, see above)